### ISSUE TITLE
Cemu doesn't register certain button events from F310 controller on macOS

### ISSUE THREAD

**Issue by alicedb2** [cemu-project/Cemu#957]
I was able to get my Logitech F310 controller to work *almost* completely with Cemu on my Mac M2 running Ventura. Cemu is running absolutely great through Rosetta 2. I first had to use this procedure https://gist.github.com/jackblk/8138827afd986f30cf9d26647e8448e1.

By *almost* I mean except for four buttons that don't registers in the input screen. Those are the two middle buttons and clicking the joysticks (see https://imgur.com/a/jATy4z0). In the input screen "Emulated controller" = "Wii U GamePad" and "Controller" = "Logitech Dual Action [SDLController]".

In Enjoyable, which is what I use right now as workaround to map them to keyboard presses and adding a Keyboard controller in Cemu, those buttons register fine:

- "Back" button (left) registers as "Button 9", 
- "Start" button (right) as "Button 10"
- Clicking the left joystick as "Button 11"
- Clicking the right joystick as "Button 12"

Goal is to be able to map them to +/-, or blow mic/show screen, or Left/Right Axis click.

**Comment by Squall-Leonhart** [comment]
the SDL implementation for the F310 is based on the Xinput mode, not the directinput mode.

**Comment by alicedb2** [comment]
Yes, in Enjoyable button numbering in the interface starts at 1, so it tracks with `SDL_gamecontrollerdb.h` where it starts at 0. I'm looking at line 403 of https://github.com/spurious/SDL-mirror/blob/master/src/joystick/SDL_gamecontrollerdb.h, and with that indexing in mind everything seems to be in order between SDL and what Enjoyable sees.

What's special about those buttons that they don't even register in Cemu while all others do?

**Comment by Squall-Leonhart** [comment]
Check with the tool inside the SDL sdk itself

https://github.com/libsdl-org/SDL/releases/tag/release-2.28.3

**Comment by alicedb2** [comment]
I can confirm that `test/testgamecontroller` tool does not pick up the back/start and joystick click events at all or incorrectly in SDL 2.28.3 and 2.28.4. It does work correctly with SDL 2.0.22, which is what the gamepad tool I used above is compled against.

**Comment by Squall-Leonhart** [comment]
You'll want to report your issue upstream then, this is not cemu's doing and it can only rely on SDL contributors to resolve

**Comment by alicedb2** [comment]
Fixed in https://github.com/libsdl-org/SDL/pull/8557, now part of the SDL main branch.

**Comment by Squall-Leonhart** [comment]
Can be closed
